compiler/rustc_errors/src/styled_buffer.rs

```diff
@@ -1,39 +1,56 @@
 // Code for creating styled buffers
 
 use crate::snippet::{Style, StyledString};
-use std::iter;
 
 #[derive(Debug)]
 pub struct StyledBuffer {
-    text: Vec<Vec<char>>,
-    styles: Vec<Vec<Style>>,
+    lines: Vec<Vec<StyledChar>>,
+}
+
+#[derive(Debug)]
+struct StyledChar {
+    chr: char,
+    style: Style,
+}
+
+impl StyledChar {
+    fn new(chr: char, style: Style) -> Self {
+        StyledChar { chr, style }
+    }
+}
+
+impl Default for StyledChar {
+    fn default() -> Self {
+        StyledChar::new(' ', Style::NoStyle)
```

Comment:
this differs from the default for `char`, is this intentional?

Reply:
i think I would prefer to add an associated constant instead, so something like
```rust
impl StyledChar {
    const SPACE: Self = StyledChar::new(' ', Style::NoStyle);
}
```
that requires you to make `StyledChar::new` a const fn

Reply:
Fixed.